auslib/admin/views/releases.py

```diff
@@ -297,12 +305,56 @@ def _delete(self, release, changed_by, transaction):
             cef_event("Bad input", CEF_WARN, errors=form.errors)
             return Response(status=400, response=json.dumps(form.errors))
 
-        dbo.releases.deleteRelease(changed_by=changed_by, name=release['name'],
-                                   old_data_version=form.data_version.data, transaction=transaction)
+        try:
+            dbo.releases.deleteRelease(changed_by=changed_by, name=release['name'],
+                                       old_data_version=form.data_version.data, transaction=transaction)
+        except ReadOnlyError as e:
+                msg = "Couldn't update release: %s" % e
+                cef_event("Bad input", CEF_WARN, errors=msg)
+                return Response(status=403, response=json.dumps({"data": e.args}))
 
         return Response(status=200)
 
 
+class ReleaseReadOnlyView(AdminView):
+    """/releases/:release/read_only"""
+
+    def get(self, release):
+        try:
+            is_release_read_only = dbo.releases.isReadOnly(name=release, limit=1)
+        except:
+            return Response(status=404,
+                            response='Requested release does not exist')
+
+        return jsonify({
+            'read_only': is_release_read_only
+        })
+
+    @requirelogin
+    @requirepermission('/releases/:name/read_only')
+    def _put(self, release, changed_by, transaction):
+        form = ReadOnlyForm()
+
+        if not form.validate() and not request.json:
```

Comment:
What's the "and not request.json" part for? I don't see it getting used again. I think WTForms handles finding data in the request; I don't think you need to touch request.json yourself.